src/main/java/at/ac/tuwien/student/sese2017/xp/hotelmanagement/web/StaffController.java

```diff
@@ -80,18 +77,49 @@ public String createCustomer(Model model) {
   @PostMapping("/staff/customer/create")
   public String postCustomer(Model model, @ModelAttribute CustomerEntity entity) {
     log.info("post customer - Page called");
+    boolean updated = false;
     try {
-      Long customerId = customerService.create(entity);
-      log.info("created customer {}", customerId);
-      model.addAttribute("note",
-          String.format("Kunde %s erfasst! (%d)", entity.getName(), customerId));
-      model.addAttribute(CUSTOMER_ATTRIBUTE_NAME, new CustomerEntity());
+      Long customerId = customerService.save(entity);
+
+      if (entity.getId() != null) {
+        log.info("Existing entity (id {}) changed", entity.getId());
+        updated = true;
+      } else {
+        log.info("created customer {}", customerId);
+        model.addAttribute("note",
+            String.format("Kunde %s erfasst! (%d)", entity.getName(), customerId));
+        model.addAttribute(CUSTOMER_ATTRIBUTE_NAME, new CustomerEntity());
+      }
     } catch (ValidationException e) {
-      model.addAttribute("note", "Fehler");
+      model.addAttribute("note", "Fehler - " + e.getMessage());
       model.addAttribute(CUSTOMER_ATTRIBUTE_NAME, entity);
+      log.warn("Customer save validation failed", e);
+    }
+    if (updated) {
+      return "redirect:/staff/search";
```

Comment:
Wieso wird hier auf die search redirected? Wie schaut hier der Workflow aus? Wenn einfach nur zur Suche redirected werden soll, weil man von dort bearbeitet hat dann sollte vielleicht auch gleich wieder nach dem Namen des Customers gesucht werden, damit er angezeigt wird und man überprüfen kann, was man geändert hat. Weiteres gibt es keine Erfolgsmeldung. Wenn man den Customer allerdings gleich in der Suche sehen könnte wäre das vielleicht gar nicht notwendig.

Reply:
gibts da schon eine passende loesung? kann ich den searchterm in der URL mitgeben?

Reply:
Du kannst als QueryParameter `domain=CUSTOMERS` und `keywords=Abbey` 

Beispiel: `staff/search?keywords=Abbey%20Fields&domain=CUSTOMERS`

Man kann alerts hinzufügen zur Search-Seite bei redirects für Feedback
```java
public String cancelReceipt(..., RedirectAttributes redir) {
  ...
  redir.addFlashAttribute("success", "Rechnung storniert.");
  ...
  return "redirect:...";
}
```
Es gibt `danger` für Fehler und `success` für Erfolgsnachrichten.

Reply:
done